ddl/table.go

```diff
@@ -71,18 +71,51 @@ func onCreateTable(d *ddlCtx, t *meta.Meta, job *model.Job) (ver int64, _ error)
 	if tbInfo.DirectPlacementOpts != nil && tbInfo.PlacementPolicyRef != nil {
 		return ver, errors.Trace(ErrPlacementPolicyWithDirectOption.GenWithStackByArgs(tbInfo.PlacementPolicyRef.Name))
 	}
+	var bundle *placement.Bundle
 	if tbInfo.DirectPlacementOpts != nil {
 		// check the direct placement option compatibility.
 		if err := checkPolicyValidation(tbInfo.DirectPlacementOpts); err != nil {
 			return ver, errors.Trace(err)
 		}
+		// build bundle from direct placement options.
+		bundle, err = placement.NewBundleFromOptions(tbInfo.DirectPlacementOpts)
+		if err != nil {
+			return ver, errors.Trace(err)
+		}
 	}
 	if tbInfo.PlacementPolicyRef != nil {
 		// placement policy reference will override the direct placement options.
-		_, err = checkPlacementPolicyExistAndCancelNonExistJob(t, job, tbInfo.PlacementPolicyRef.ID)
+		po, err := checkPlacementPolicyExistAndCancelNonExistJob(t, job, tbInfo.PlacementPolicyRef.ID)
 		if err != nil {
 			return ver, errors.Trace(infoschema.ErrPlacementPolicyNotExists.GenWithStackByArgs(tbInfo.PlacementPolicyRef.Name))
 		}
+		// build bundle from placement policy.
+		bundle, err = placement.NewBundleFromOptions(po.PlacementSettings)
+		if err != nil {
+			return ver, errors.Trace(err)
+		}
+	}
+	if bundle == nil {
+		// get the default bundle from DB or PD.
+		bundle = infoschema.GetBundle(d.infoCache.GetLatest(), []int64{schemaID})
```

Comment:
Why not generating the default bundle using the schema's meta.

Reply:
yes we can, copyed from wanghe's work for simplicity.
cases is that: once schema did not set a policy, here may need a default bundle with `PDBundleID` (also from cache or via http to pd). 

cache problem may be your concern, I will change this when all to-do things is much more clear.